File: src/tutorial/java/ch/softappeal/yass/tutorial/contract/Instrument.java
package ch.softappeal.yass.tutorial.contract;

import ch.softappeal.yass.util.Check;
import ch.softappeal.yass.util.Tag;

public abstract class Instrument {

    @Tag(1) public final int id;
    @Tag(2) public final String name;

    protected Instrument(final int id, final String name) {
        this.id = id;
        this.name = Check.notNull(name);
    }

}

File: kotlin/tutorial/main/ch/softappeal/yass/tutorial/contract/instrument.kt
package ch.softappeal.yass.tutorial.contract.instrument

import ch.softappeal.yass.remote.OneWay
import ch.softappeal.yass.tutorial.contract.Expiration
import ch.softappeal.yass.tutorial.contract.Instrument

interface InstrumentService {
    val instruments: List<Instrument>
    /** This method does nothing meaningful. It just shows how to make [OneWay] method calls. */
    @OneWay
    fun showOneWay(testBoolean: Boolean, testInt: Int)
}

class Bond(id: Int, name: String, val coupon: Double, val expiration: Expiration) : Instrument(id, name)
